File: git_vs/1920번 수 찾기.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> narray;
int n;

int binarySearch(int key) {
	int l = 0, r = n - 1;
	int mid = (l + r) / 2;
	while (l <= r) {
		if (narray[mid] < key) {
			l = mid + 1;
			mid = (l + r) / 2;
		}
		else if(narray[mid]>key){
			r = mid - 1;
			mid = (l + r) / 2;
		}
		else if(narray[mid] == key)return 1;
	}
	return 0;
}

int main() {
	cin.tie(0);
	ios::sync_with_stdio(false);
	cin >> n;
	for (int i = 0; i < n; i++) {
		int tmp;
		cin >> tmp;
		narray.push_back(tmp);
	}
	sort(narray.begin(), narray.end());
	int m;
	cin >> m;
	for (int i = 0; i < m; i++) {
		int tmp;
		cin >> tmp;
		cout << binarySearch(tmp)<<"\n";
	}
	return 0;
}

File: git_vs/1920번 수 찾기.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 29 16:18:32 2020

@author: user
"""

import sys
input=sys.stdin.readline

def binarySearch(n, narray, key):
    l=0
    r=n-1
    mid=(l+r)//2
    while l<=r:
        if narray[mid]<key:
            l=mid+1
            mid=(l+r)//2
        elif narray[mid]>key:
            r=mid-1
            mid=(l+r)//2
        else:
            print(1)
            return
    print(0)
    return 

if __name__=="__main__":
    n=int(input())
    narray=list(map(int,(input().strip()).split()))
    narray.sort()
    m=int(input())
    marray=list(map(int,input().split()))
    for i in range(m):
        tmp=marray[i]
        binarySearch(n, narray, tmp)
